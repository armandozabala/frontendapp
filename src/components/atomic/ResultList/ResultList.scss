@import "../../../styles/main.scss";

.meli-results-list {
  border-radius: $results-list-radius;
  padding: $results-list-padding;
  background-color: $results-list-background-color;
  margin: $results-list-padding 0;

  & > .meli-grid {
    & > div {
      & + div {
        margin-top: $results-list-padding;

        & > * {
          border-top: $results-list-border;
          padding-top: $results-list-padding;
        }
      }
    }
  }
}
