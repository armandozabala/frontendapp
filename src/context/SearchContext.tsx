import { createContext } from "react";

export type SearchContextType = {
  search: string;
  setSearch: Function;
};

export const SearchContext = createContext<SearchContextType>();
