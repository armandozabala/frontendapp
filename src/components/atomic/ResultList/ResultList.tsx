import React from "react";
import ResultsListItem from "../ResultsListItem/ResultsListItem";

import "./ResultList.scss";

export const ResultsList = ({ items }: any) => {
  return (
    <section className="meli-results-list">
      <div className="meli-grid meli-child-width-1-1">
        {items.length ? (
          items.map((item: any, index: any) => (
            <div key={index}>
              <ResultsListItem data={item} />
            </div>
          ))
        ) : (
          <div>Sin resultados. Prueba buscar otra cosa.</div>
        )}
      </div>
    </section>
  );
};

export default ResultsList;
