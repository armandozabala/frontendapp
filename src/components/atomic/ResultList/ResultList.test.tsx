import React from "react";
import ResultList from "./ResultList";
import { mount } from "enzyme";
import { SearchContext } from "../../../context/searchContext";
import { MemoryRouter } from "react-router-dom";

describe("ResultList", () => {
  test("ResultList", () => {
    const items: any = [
      {
        id: "MLA829590869",
        title: "Mochila de madera con cierre de bot√≥n",
        price: {
          currency: "ARS",
          amount: 1296,
          decimals: "99",
        },
        picture:
          "http://mla-s2-p.mlstatic.com/829590869-MLA3100827091_062019-I.jpg",
      },
    ];

    const wrapper = mount(
      <MemoryRouter>
        <ResultList items={items} />{" "}
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("ResultList empty", () => {
    const items: any = [];

    const wrapper = mount(
      <MemoryRouter>
        <ResultList items={items} />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
