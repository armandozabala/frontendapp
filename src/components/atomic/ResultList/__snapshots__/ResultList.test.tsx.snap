// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultList ResultList 1`] = `ReactWrapper {}`;

exports[`ResultList ResultList empty 1`] = `ReactWrapper {}`;
